Gems/Atom/Asset/Shader/Code/Source/Editor/ShaderVariantListBuilder.cpp

```diff
@@ -0,0 +1,586 @@
+            }
+
+
+            if (shouldExitEarlyFromProcessJob)
+            {
+                // Let's create a fake job that will succeed and create nothing.
+                // This only happens with shader variant list files provided by the engine,
+                // but are overriden by the game project.
+                AssetBuilderSDK::JobDescriptor jobDescriptor;
+                
+                jobDescriptor.m_priority = -5000;
+                jobDescriptor.m_critical = false;
+                // We are going to create a fake job that we know will fail.
+                // We need to use a job key of a real product (In this case we are using the job key for the ShaderVariantTreeAsset).
+                // Using a real product job key will guarantee to clear old errors in the future, because a successful product build will
+                // replace the lingering error thanks to matching job keys.
+                jobDescriptor.m_jobKey = JobKey;
+                jobDescriptor.SetPlatformIdentifier(AssetBuilderSDK::CommonPlatformName);
+         
+                // The value doesn't matter, what matters is the presence of the key which will
+                // signal that no assets should be produced on behalf of this shadervariantlist because
+                // the game project overrode it.
+                jobDescriptor.m_jobParameters.emplace(ShouldExitEarlyFromProcessJobParam, variantListFullPath);
+                
+                response.m_createJobOutputs.push_back(jobDescriptor);
+
+                response.m_result = AssetBuilderSDK::CreateJobsResultCode::Success;
+                return;
+            }
+
+            // TODO. This work will be done as another PR after this work is merged.
+            // See if there are other *.shadervariantlist files that are related with this one, and add them
+            // as source dependency.
+
+            // This is the real job
+            {
+                AssetBuilderSDK::JobDescriptor jobDescriptor;
+            
+                jobDescriptor.m_priority = 1;
+                jobDescriptor.m_critical = false;
+            
+                jobDescriptor.m_jobKey = JobKey;
+                jobDescriptor.SetPlatformIdentifier(AssetBuilderSDK::CommonPlatformName);
+
+                jobDescriptor.m_jobParameters.emplace(ShaderVariantListAbsolutePathJobParam, variantListFullPath);
+                jobDescriptor.m_jobParameters.emplace(ShaderAbsolutePathJobParam, shaderSourceFileAbsolutePath);
+            
+                response.m_createJobOutputs.push_back(jobDescriptor);
+            }
+
+            response.m_result = AssetBuilderSDK::CreateJobsResultCode::Success;
+        }  // End of CreateJobs
+
+
+        // This function is temporarily needed until the AssetSystem APIs provides
+        // an API to locate intermediate assets.
+        static AZStd::string GetAbsolutePathOfIntermediateAsset(const AZStd::string sourceShaderVariantListAbsolutePath)
```

Comment:
```suggestion
        static AZStd::string GetAbsolutePathOfIntermediateAsset(const AZStd::string& sourceShaderVariantListAbsolutePath)
```

Reply:
Function was removed in the last commit.